activesupport/lib/active_support/error_reporter.rb

```diff
@@ -108,13 +110,31 @@ def handle(*error_classes, severity: :warning, context: {}, fallback: nil, sourc
     #   source of the error. Subscribers can use this value to ignore certain
     #   errors. Defaults to <tt>"application"</tt>.
     def record(*error_classes, severity: :error, context: {}, source: DEFAULT_SOURCE)
-      error_classes = [StandardError] if error_classes.blank?
+      error_classes = DEFAULT_RESCUE if error_classes.empty?
       yield
     rescue *error_classes => error
       report(error, handled: false, severity: severity, context: context, source: source)
       raise
     end
 
+    # Evaluates the given block, reporting any unhandled error.
+    # The error is then either re-raised if +config.consider_all_requests_local == true+
+    # or otherwise swallowed.
+    #
+    # In other words in behaves like +record+ in development and test, and like +handle+
+    # in production.
+    def debug(*error_classes, severity: :warning, context: {}, fallback: nil, source: DEFAULT_SOURCE)
+      error_classes = DEFAULT_RESCUE if error_classes.empty?
+      yield
+    rescue *error_classes => error
+      report(error, handled: true, severity: severity, context: context, source: source)
+      if @debug_mode
+        raise
+      else
+        fallback.call if fallback
```

Comment:
Any reason to pass `error` to the fallback? Or would it be better to wait for someone to come up with a reasonable use-case where fallback may need to know about the error instance?

Reply:
The problem is we pass it, is that all callers must accept it, so if the fallback is a lambda or method, we'd break existing code.

I think in the vast majority of the case you don't need it. If you do you can use a "manual" `begin/rescue/Rails.error.report`.

Reply:
Or we'd need to do an arity check, but that's pushing it. So yeah, might as well wait for a demand. I'd rather not make this API too large.